spec/async_producer_spec.rb

```diff
@@ -33,6 +33,24 @@ def instrument(name, payload = {})
     )
   }
 
+  describe "#initialize" do
+    it "creates a unique Mutex for each instance" do
+      async_producer2 = Kafka::AsyncProducer.new(
+        sync_producer: sync_producer,
+        instrumenter: instrumenter,
+        logger: logger,
+      )
+
+      [async_producer, async_producer2].each do |p|
+        class << p
+          attr_reader :thread_mutex
+        end
+      end
+
+      expect(async_producer.thread_mutex).to_not eq(async_producer2.thread_mutex)
+    end
+  end
```

Comment:
I don't think we need a test for this – it's not testing functionality, but rather implementation details.

Reply:
Fixed up in 5f22fbd2bc46af22b10e15ead966e5651cc47d2d